core/kazoo_number_manager/src/knm_tasks.erl

```diff
@@ -0,0 +1,206 @@
+                        ,{<<"mandatory">>, [<<"number">>
+                                           ,<<"account_id">>
+                                           ]}
+                        ,{<<"optional">>, [<<"auth_by">>
+                                          ]}
+                        ])
+     }
+
+    ,{<<"delete">>
+     ,kz_json:from_list([{<<"description">>, <<"Bulk-remove numbers">>}
+                        ,{<<"expected_content">>, <<"text/csv">>}
+                        ,{<<"mandatory">>, [<<"number">>
+                                           ]}
+                        ,{<<"optional">>, [<<"auth_by">>
+                                          ]}
+                        ])
+     }
+
+    ,{<<"reserve">>
+     ,kz_json:from_list([{<<"description">>, <<"Bulk-move numbers to reserved (adding if missing)">>}
+                        ,{<<"expected_content">>, <<"text/csv">>}
+                        ,{<<"mandatory">>, [<<"number">>
+                                           ,<<"account_id">>
+                                           ]}
+                        ,{<<"optional">>, [<<"auth_by">>
+                                          ]}
+                        ])
+     }
+
+    ,{<<"add">>
+     ,kz_json:from_list([{<<"description">>, <<"Bulk-create numbers">>}
+                        ,{<<"expected_content">>, <<"text/csv">>}
+                        ,{<<"mandatory">>, [<<"number">>
+                                           ,<<"account_id">>
+                                           ]}
+                        ,{<<"optional">>, [<<"auth_by">>
+                                          ,<<"module_name">>
+                                          ]}
+                        ])
+     }
+
+    %% ,{<<"update_features">>
+    %%  ,kz_json:from_list([{<<"description">>, <<"Bulk-update features of numbers">>}
+    %%                     ,{<<"expected_content">>, <<"text/csv">>}
+    %%                     ,{<<"mandatory">>, [<<"number">>
+    %%                                        ]}
+    %%                     ,{<<"optional">>, [<<"cnam.inbound">>
+    %%                                       ,<<"cnam.outbound">>
+    %%                                       ,<<"e911.street_address">>
+    %%                                            %%TODO: exhaustive list
+    %%                                       ]}
+    %%                     ])
+    %%  }
+    ].
+
+%% Verifiers
+
+-spec number(ne_binary()) -> boolean().
+number(<<"+", _/binary>>) -> 'true';
+number(_) -> 'false'.
+
+-spec account_id(ne_binary()) -> boolean().
+account_id(?MATCH_ACCOUNT_RAW(_)) -> 'true';
+account_id(_) -> 'false'.
+
+-spec auth_by(api_binary()) -> boolean().
+auth_by('undefined') -> 'true';
+auth_by(?MATCH_ACCOUNT_RAW(_)) -> 'true';
+auth_by(_) -> 'false'.
+
+-spec module_name(api_binary()) -> boolean().
+module_name('undefined') -> 'true';
+module_name(Thing) ->
+    Modules = [<<"knm_bandwidth2">>
```

Comment:
I worry this will break people who have built their own carrier modules. I wonder if you could look at knm modules and find those with `-behaviour(knm_gen_carrier).` in them? Maybe build a module, like kz_mime, that has valid carrier modules?